pkg/services/serviceaccounts/manager/service.go

```diff
@@ -125,10 +144,62 @@ func (sa *ServiceAccountsService) CreateServiceAccount(ctx context.Context, orgI
 	return sa.store.CreateServiceAccount(ctx, orgID, saForm)
 }
 
-func (sa *ServiceAccountsService) DeleteServiceAccount(ctx context.Context, orgID, serviceAccountID int64) error {
-	return sa.store.DeleteServiceAccount(ctx, orgID, serviceAccountID)
+func (sa *ServiceAccountsService) RetrieveServiceAccount(ctx context.Context, orgID int64, serviceAccountID int64) (*serviceaccounts.ServiceAccountProfileDTO, error) {
+	return sa.store.RetrieveServiceAccount(ctx, orgID, serviceAccountID)
 }
 
 func (sa *ServiceAccountsService) RetrieveServiceAccountIdByName(ctx context.Context, orgID int64, name string) (int64, error) {
 	return sa.store.RetrieveServiceAccountIdByName(ctx, orgID, name)
 }
+
+func (sa *ServiceAccountsService) DeleteServiceAccount(ctx context.Context, orgID, serviceAccountID int64) error {
```

Comment:
I would add checks for these methods in Service to check for `orgID = 0` or `serviceAccountID=0` or `apiKeyID=0` and return an `ErrInvalid... ` to ensure the ABI is not misused

Reply:
added validation in commit: b8b8b7fc9a1bb990820f415362d2cc3ae835d762